### ISSUE TITLE
magiskpolicy: Error when creating a new domain

### ISSUE THREAD

**Issue by IgorEisberg** [topjohnwu/Magisk#3071]
Hey mate,
Maybe I'm misunderstanding the usage here, but you might know better.
I'm trying to port a recent version of Dolby Atmos, which requires modifying the sepolicy.
I can't seem to be able to add some new domains which are required to add the needed rules due to an error:

> dipper:/ $ su
> dipper:/ # magiskpolicy --live "create hal_dms_default"
> Load policy from: /sys/fs/selinux/policy
> Error in: create hal_dms_default

Same error in case of directly modifying a precompiled_sepolicy file:

> $ ./magiskpolicy --load precompiled_sepolicy --save precompiled_sepolicy "create hal_dms_default"
> Load policy from: precompiled_sepolicy
> Error in: create hal_dms_default

If using sepolicy-inject using the equivalent command, on a precompiled_sepolicy file directly, I get a "Segmentation fault".

> $ ./sepolicy-inject -Z hal_dms_default -P precompiled_sepolicy
> Loaded policy: 1 users, 4 roles, 2301 types, 0 bools, 1 sens, 1024 cats, 97 classes, 43483 rules, 0 cond rules
> Segmentation fault

This happens with any precompiled sepolicy file I threw at it, even one from Oreo.

**Comment by topjohnwu** [comment]
```
One policy statement should be treated as one parameter;
this means each policy statement should be enclosed in quotes.
Multiple policy statements can be provided in a single command.

Statements has a format of "<rule_name> [args...]".
Arguments labeled with (^) can accept one or more entries. Multiple
entries consist of a space separated list enclosed in braces ({}).
Arguments labeled with (*) are the same as (^), but additionally
support the match-all operator (*).

Example: "allow { s1 s2 } { t1 t2 } class *"
Will be expanded to:

allow s1 t1 class { all-permissions-of-class }
allow s1 t2 class { all-permissions-of-class }
allow s2 t1 class { all-permissions-of-class }
allow s2 t2 class { all-permissions-of-class }

Supported policy statements:

"allow *source_type *target_type *class *perm_set"
"deny *source_type *target_type *class *perm_set"
"auditallow *source_type *target_type *class *perm_set"
"dontaudit *source_type *target_type *class *perm_set"

"allowxperm *source_type *target_type *class operation xperm_set"
"auditallowxperm *source_type *target_type *class operation xperm_set"
"dontauditxperm *source_type *target_type *class operation xperm_set"
- The only supported operation is 'ioctl'
- xperm_set format is either 'low-high', 'value', or '*'.
  '*' will be treated as '0x0000-0xFFFF'.
  All values should be written in hexadecimal.

"permissive ^type"
"enforce ^type"

"typeattribute ^type ^attribute"

"type type_name ^(attribute)"
- Argument 'attribute' is optional, default to 'domain'

"attribute attribute_name"

"type_transition source_type target_type class default_type (object_name)"
- Argument 'object_name' is optional

"type_change source_type target_type class default_type"
"type_member source_type target_type class default_type"

"genfscon fs_name partial_path fs_context"
```

**Comment by IgorEisberg** [comment]
Thanks, apparently I had to use the Canary release, because "type" statement isn't available in 20.4.
